Cookbook/Proposals/SmarterDiffs.md

```diff
@@ -0,0 +1,97 @@
+```
+
+instead of
+
+```
+expected to equal <[State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: false, shouldShowTestAndKitsBanner: false, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.idle, isInsuranceValid: true), State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: false, shouldShowTestAndKitsBanner: false, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.loading, isInsuranceValid: true), State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: true, shouldShowTestAndKitsBanner: true, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.loaded, isInsuranceValid: true)]>, got <[State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: false, shouldShowTestAndKitsBanner: false, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.idle, isInsuranceValid: true), State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: false, shouldShowHealthcheckBanner: false, shouldShowTestAndKitsBanner: false, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.loading, isInsuranceValid: true), State(firstName: "Brienne", lastName: "Skrilla", shouldShowNHSBanner: true, shouldShowHealthcheckBanner: true, shouldShowTestAndKitsBanner: true, shouldShowGPAtHandStatusCard: false, appointments: [], prescriptions: [], partnerLogoURL: nil, partnerLogoCaption: "", status: Babylon.HomeViewModelV2.Status.loaded, isInsuranceValid: true)]>
+```
+
+As can be seen from the diff it does not show in what exact array item the failure happened. This is something that we will need to adjust along with [this issue](https://github.com/krzysztofzablocki/Difference/issues/1). 
+
+Proposed way to integrate this feature is to copy the implementation (with proper author attribution) and adjust it to our needs, optionally we can make pull requests to the original library.
+
+## Impact on existing codebase
+
+This approach can be rolled out gradually and used for new test (Xcode templates will use a new assert) or tests can be refactored in batches by replacing existing assertions with a new assert.
+
+## Alternatives considered
+
+As an alternative we can enforce asserting on individual states instead of all states which will make diffs less of an issue. But as mentioned before it will not eliminate it.
```

Comment:
I like the idea of seeing all the states. It makes it a little bit easier to figure out how given view model works. Plus, those tests are already a bit broad, so I'd say it's fine. Definitely agree with downsides you pointed out in "Option 1 - assert states one by one".